src/core/thread/mle.cpp

```diff
@@ -1061,7 +1061,9 @@ otError Mle::AppendLeaderData(Message &aMessage)
 
 void Mle::FillNetworkDataTlv(NetworkDataTlv &aTlv, bool aStableOnly)
 {
-    uint8_t length;
+    uint8_t length = 255; // sizeof( NetworkDataTlv::mNetworkData )
```

Comment:
I'll go for this sizeof arithmetic, though `kMaxSize` might be lesser evil.

Whenever I have to rely on a stuff like sizeof of packed nested classes, my intuition screams to me that it is going to backfire sooner or later, e.g. some windows compiler using strange alignment rules or whatever. So, I do not like to write code of when I can not guarantee (or even test on all platforms) its correctness.